tools/ports/__init__.py

```diff
@@ -64,6 +65,22 @@ def load_port(name):
     port_variants[variant] = (port.name, extra_settings)
 
 
+def load_port_by_name(name):
+  port = __import__(name, globals(), level=1, fromlist=[None])
+  init_port(name, port)
+
+
+def load_port_by_path(path):
+  name = os.path.splitext(os.path.basename(path))[0]
+  module_name = f'tools.ports.{name}'
```

Comment:
Does this have to be globally unique?   i.e. what happens if we have `--use-port=foo/port.py --use-port=bar/port.py` .. currenetly they would both result in `tools.ports.port` here.. is that OK?

Reply:
For the sake of supporting the use case of being able to override a built-in port, meaning I should be able to write `--use-port=locked_version_ports/sdl2.py` then I feel like yes we should support this. 

Here is my thinking:
* I feel like ports will have relatively unique name in general (they should probably not be named "port" obviously), so the likelihood of conflicts is pretty low
* If you want we could display a warning message when such a use is detected

Reply:
What do you mean by supporting `--use-port=locked_version_ports/sdl2.py`.. does that do something special because it shadows `--use-port=sdl2`?   When is that useful?

Reply:
It is clear that the old (`-sUSE_SDL2`) syntax vs new syntax (`--use-port=sdl2`) makes everything so much more complicated.

Implemeting "disallow aliasing of builtin ports" but allowing for other ports does not achieve easy locking down version and makes the code far more complicated.

I suppose it is possible to copy `sld2.py` in my project and call it `sld2_local.py` and that will do the trick since the name won't collide and it makes it clear that I am not using the built-in one accidentally...

I will disallow aliasing entirely then.

Reply:
No. I thought we decided not to allow aliasing at all. I implemented it this way:

```python
  name = os.path.splitext(os.path.basename(path))[0]
  if name in ports_by_name:
    utils.exit_with_error(f'port path [{path}] is invalid: duplicate port name {name}')
```

Reply:
On nice!  I didn't see that.